language-plutus-core/src/Language/PlutusCore/MkPlc.hs

```diff
@@ -85,23 +87,31 @@ embed = \case
     Unwrap a t        -> unwrap a (embed t)
     IWrap a ty1 ty2 t -> iWrap a ty1 ty2 (embed t)
 
--- | A "variable declaration", i.e. a name annnd a type for a variable.
+-- | A "variable declaration", i.e. a name and a type for a variable.
 data VarDecl tyname name ann = VarDecl
     { varDeclAnn  :: ann
     , varDeclName :: name ann
     , varDeclType :: Type tyname ann
-    } deriving (Functor, Show, Eq, Generic)
+    } deriving (Functor, Show, Generic)
+
+instance HasUniques (Term tyname name ann) => Eq (VarDecl tyname name ann) where
```

Comment:
I'm not quite sure why we have these instances.

Reply:
Not sure either. They don't seem to be needed. Should I remove them?

Reply:
They make sense, so I'll keep them for now.